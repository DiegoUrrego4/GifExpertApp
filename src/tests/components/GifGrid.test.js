import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme/build';
import { useFetchGifs } from '../../hooks/useFetchGifs';
import GifGrid from '../../components/GifGrid';

jest.mock('../../hooks/useFetchGifs');

describe('Tests in GifGrid', () => {
  const category = 'Dragon Ball';

  test('The component must be displayed correctly', () => {
    // ! Este es EL valor inicial que tiene nuestro estado en useFetchGifs
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true,
    });
    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('Must display  when loading images useFetchGifs', () => {
    const gifs = [
      {
        id: 'ABC',
        url: 'https://localhost/cualquier/cosa.jpg',
        title: 'Cualquier Cosa',
      },
      {
        id: 'ABCD',
        url: 'https://localhost/cualquier/otraCosa.jpg',
        title: 'Cualquier Otra Cosa',
      },
    ];
    // ! Al traer "informaci√≥n" nuestro estado debe cambiar, tal cual como lo hace nuestro componente
    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false,
    });
    const wrapper = shallow(<GifGrid category={category} />);
    // expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('p').exists()).toBe(false);
    expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
  });
});
