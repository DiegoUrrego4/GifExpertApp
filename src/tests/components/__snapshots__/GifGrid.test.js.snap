// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in GifGrid The component must be displayed correctly 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeInRight"
  >
    Dragon Ball
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
